src/analytics/formTracker/formTracker.ts

```diff
@@ -1,6 +1,182 @@
+
+      if (
+        element.type === "hidden" ||
+        element.type === "fieldset" ||
+        element.type === "submit"
+      ) {
+        continue;
+      }
+
+      if (
+        (element.type === "radio" || element.type === "checkbox") &&
+        element.checked
+      ) {
+        selectedFields.push({
+          id: element.id,
+          name: element.name,
+          value: document.querySelector(`label[for="${element.id}"]`)
+            ?.textContent
+            ? document
+                .querySelector(`label[for="${element.id}"]`)
+                ?.textContent?.trim()
+            : "undefined",
+          type: element.type,
+        });
+      } else if (element.type === "textarea" || element.type === "text") {
+        selectedFields.push({
+          id: element.id,
+          name: element.name,
+          value: element.value,
+          type: element.type,
+        });
+      } else if (element.type === "select-one") {
+        const selectedElement = form.elements[i] as HTMLSelectElement;
+        selectedFields.push({
+          id: selectedElement.id,
+          name: selectedElement.name,
+          value: selectedElement.options[selectedElement.selectedIndex].text,
+          type: selectedElement.type,
+        });
+      }
+    }
+    return selectedFields;
+  }
+
+  /**
+   * Returns the field type based on the given FormField array.
+   *
+   * @param {FormField[]} elements - An array of FormField objects.
+   * @return {string} The field type based on the elements.
+   */
+  getFieldType(elements: FormField[]): string {
+    if (elements[0].type === "textarea" || elements[0].type === "text") {
+      return "free text field";
+    } else if (elements[0].type === "select-one") {
+      return "drop-down list";
+    } else {
+      return elements[0].type;
+    }
+  }
+
+  /**
+   * Retrieves the label of a field.
+   *
+   * @return {string} The label of the field.
+   */
+  getFieldLabel(): string {
+    let labels: HTMLCollectionOf<HTMLLegendElement | HTMLLabelElement> =
+      document.getElementsByTagName("legend");
+    if (!labels.length) {
+      labels = document.getElementsByTagName("label");
+    }
+    let label: string = "";
+    for (let i = 0; i < labels.length; i++) {
+      if (labels[i].textContent) {
+        label += labels[i]?.textContent?.trim();
+        if (i > 1 && i < labels.length - 1) {
+          label += ", ";
+        }
+      }
+    }
+    return label;
+  }
+
+  /**
+   * Retrieves the field value from an array of form fields.
+   *
+   * @param {FormField[]} elements - The array of form fields.
+   * @return {string} The concatenated field values separated by a comma (if there is more than one field).
+   */
+  getFieldValue(elements: FormField[]): string {
+    let value = "";
+    const separator = elements.length > 1 ? ", " : "";
+    elements.forEach((element) => {
+      value += element.value + separator;
```

Comment:
do we need to add element?.value incase it doesn't exist?

Reply:
You will always have a value because you will always elements.
This function is called only if the code has found fields into the form (line 36) and into getFields, if we don't have value, we populate the property with "undefined".